### ISSUE TITLE
Adapt bert-as-service for QA

### ISSUE THREAD

**Issue by eugenekoh** [hanxiao/bert-as-service#179]
**Prerequisites**

> Please fill in by replacing `[ ]` with `[x]`.

* [ ] Are you running the latest `bert-as-service`?
* [ ] Did you follow [the installation](https://github.com/hanxiao/bert-as-service#install) and [the usage](https://github.com/hanxiao/bert-as-service#usage) instructions in `README.md`?
* [X] Did you check the [FAQ list in `README.md`](https://github.com/hanxiao/bert-as-service#speech_balloon-faq)?
* [X] Did you perform [a cursory search on existing issues](https://github.com/hanxiao/bert-as-service/issues)?

**System information**

> Some of this information can be collected via [this script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh).

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.10
- TensorFlow installed from (source or binary): source
- TensorFlow version: 1.12
- Python version: 3.6.7
- `bert-as-service` version:  
- GPU model and memory: 
- CPU model and memory:

---

### Description

Hi Han Xiao, 

I have finetuned BERT for question answering on the SQuAD dataset with run_squad.py from the BERT repository. I was wondering if it is possible to adapt bert-as-service for a question answering task. If it is possible, could I have some guidance on the steps needed to adapt it? 

Thank you.

Eugene
...

**Comment by Accentax** [comment]
Hi Eugene,

We are also working on just that. These are the steps we have been doing. 

1.  First step would be to change the graph.py to the new model, since the output are different. 
2. Then in the server init file you need to change the input function and model function to reflect the new model, features and outputs. Most of these functions can be directly imported from run_squad in the bert repo such at tokenization. 
3. The tricky part might be to pass the result to the client side, since we are not passing just a tensor but spans of text. The easiest(hacky?) way might just be to pass the output logits, and then parse the output on the client side. This would require you to re-tokenize the text and map the logits to the correct span in the text.

**Comment by eugenekoh** [comment]
Hi Alexander, 

Thanks for the reply! I'll give it a try with your guidelines.

Eugene

**Comment by erichare** [comment]
HI @Accentax @eugenekoh , I was wondering if either of you have made progress in this task and are willing to share some of the code if so?

**Comment by Accentax** [comment]
@erichare We ended up using the huggingface implementation instead. But the steps above should be relatively easy to implement. Alternatively you can look at the huggingface implementation, which is a much easier way to get started with the different squad and other model heads.
